From: George Kashperko <george@znau.edu.ua>

Modify irqflag routine to reuse tpsflag value rather than
reading TPSFLAG register once again. Change default "not
supported" value from 0x3F to -1.
Signed-off-by: George Kashperko <george@znau.edu.ua>
---
 drivers/ssb/driver_mipscore.c |    2 +-
 drivers/ssb/ssb_sb.c          |    4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)
--- linux-3.3.8.orig/drivers/ssb/driver_mipscore.c	2012-10-28 12:11:56.586324647 +0200
+++ linux-3.3.8/drivers/ssb/driver_mipscore.c	2012-10-28 12:12:03.976324139 +0200
@@ -86,7 +86,7 @@ unsigned int ssb_mips_irq33k(struct ssb_
 	unsigned int irq;
 
 	irqflag = ssb_irqflag(dev);
-	if (irqflag == 0x3f)
+	if (irqflag == -1)
 		return 6;
 	ipsflag = ssb_read32(bus->mipscore.dev, SSB_IPSFLAG);
 	for (irq = 1; irq <= 4; irq++) {
--- linux-3.3.8.orig/drivers/ssb/ssb_sb.c	2012-10-28 12:11:11.938327700 +0200
+++ linux-3.3.8/drivers/ssb/ssb_sb.c	2012-10-28 12:12:03.977324140 +0200
@@ -193,10 +193,10 @@ static u32 ssb_irqflag_sb(struct ssb_dev
 {
 	u32 tpsflag = ssb_read32(dev, SSB_TPSFLAG);
 	if (tpsflag)
-		return ssb_read32(dev, SSB_TPSFLAG) & SSB_TPSFLAG_BPFLAG;
+		return tpsflag & SSB_TPSFLAG_BPFLAG;
 	else
 		/* not irq supported */
-		return 0x3f;
+		return -1;
 }
 #endif /* CONFIG_SSB_DRIVER_MIPS */
 
